### PR TITLE
Refactor round logging; replace remaining toString calls

### PR DESCRIPTION
**Pull request by lebdron**
### Description of the Change
- Log `Round` objects by calling `toString`
- Remove remaining explicit `toString` calls for logging in `irohad`
- Fix round log in on demand ordering init

### Benefits
Simpler code

### Possible Drawbacks 
None?

### CODE REVIEW

File: irohad/main/impl/on_demand_ordering_init.cpp [hunk 1]
```diff
@@ -132,9 +132,11 @@ namespace iroha {
           // with number of peers
           auto &peer =
               current_peers_[permutation[reject_round % permutation.size()]];
-          log_->debug("For round {}, using OS on peer: {}",
-                      consensus::Round{current_round.block_round, reject_round},
-                      peer->toString());
+          log_->debug(
+              "For {}, using OS on peer: {}",
+              consensus::Round{current_round.block_round + block_round_advance,
```
Comment (Akvinikym) [inline irohad/main/impl/on_demand_ordering_init.cpp]: I thought, isn't it wasteful to create a struct just to place it into the log? Maybe do a manual output here?

File: irohad/ordering/impl/on_demand_ordering_service_impl.cpp [hunk 2]
```diff
@@ -97,9 +90,8 @@ OnDemandOrderingServiceImpl::onRequestProposal(consensus::Round round) {
   // read lock
   std::shared_lock<std::shared_timed_mutex> guard(lock_);
   auto proposal = proposal_map_.find(round);
-  log_->debug("onRequestProposal, round[{}, {}], {}returning a proposal.",
-              round.block_round,
-              round.reject_round,
+  log_->debug("onRequestProposal, {}, {}returning a proposal.",
```
Comment (Akvinikym) [inline irohad/ordering/impl/on_demand_ordering_service_impl.cpp]:
```suggestion
  log_->debug("onRequestProposal, {}, {} returning a proposal.",
```

Comment (lebdron) [reply irohad/ordering/impl/on_demand_ordering_service_impl.cpp]: https://github.com/hyperledger/iroha/pull/1871#discussion_r238148963

Comment (MBoldyrev) [reply irohad/ordering/impl/on_demand_ordering_service_impl.cpp]: :laughing: maybe we need to place a comment warning that the whitespace is not missing there! To see why, please see the substitution for these braces.

File: irohad/main/impl/on_demand_ordering_init.cpp (hunk 1 above)
Comment (lebdron) [reply irohad/main/impl/on_demand_ordering_init.cpp]: I think we can use this approach for simplicity now, and optimize it later if it turns out to be a bottleneck.

**Pull request merged**